crypto/o_time.c

```diff
@@ -358,3 +358,19 @@ static void julian_to_date(long jd, int *y, int *m, int *d)
     *m = j + 2 - (12 * L);
     *y = 100 * (n - 49) + i + L;
 }
+
+
+int OPENSSL_timegm(struct tm *tm, time_t *t)
+{
+    /* Figures it out without mucking the environment */
+    long basetime, currtime;
+
+    basetime = date_to_julian(1970, 1, 1);
+    currtime = date_to_julian(tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday);
+
+    if (t != NULL)
+        *t = (((currtime - basetime) * 24 + tm->tm_hour) * 60 + tm->tm_min) * 60 + tm->tm_sec;
```

Comment:
@richsalz wrote on Feb 17, 2017:

>time_t is unambiguous. By saying time_t is the only interface, we have only a portable unambiguous interface, and anything else should be left to the application. There's a reason why OpenSSL does things internally with ASN1 times.

I am trying to provide that interface, to extract a time_t from an ASN1_TIME, but now I am getting conflicting messages.

There _needs_ to be a mechanism to extract the time portably from an ASN1_TIME without requiring each application to write their own parse code. As quoted above, time_t is that unambiguous interface. What do you propose instead?